gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/handler/HttpGremlinResponseStreamDecoder.java

```diff
@@ -79,8 +79,8 @@ protected void decode(ChannelHandlerContext ctx, DefaultHttpObject msg, List<Obj
             ByteBuf content = ((HttpContent) msg).content();
             Attribute<Integer> bytesRead = ctx.channel().attr(BYTES_READ);
             bytesRead.set(bytesRead.get() + content.readableBytes());
-            if (bytesRead.get() > maxContentLength) {
-                throw new TooLongFrameException("Response exceeded " + maxContentLength + " bytes.");
+            if (maxResponseContentLength > 0 && bytesRead.get() > maxResponseContentLength) {
```

Comment:
We are mixing types here which could cause problems. `bytesRead.get()` is an `Integer` while `maxResponseContentLength` is a `double`. Any `maxResponseContentLength` greater than `Integer.MAX_VALUE` will never be hit.

Reply:
Whoops, thanks for catching.

Reply:
Changed to store the bytes read as Double instead of Integer.